src/main/java/com/hackathonorganizer/hackathonwriteservice/hackathon/controller/HackathonController.java

```diff
@@ -0,0 +1,48 @@
+package com.hackathonorganizer.hackathonwriteservice.hackathon.controller;
+
+import com.hackathonorganizer.hackathonwriteservice.hackathon.model.Hackathon;
+import com.hackathonorganizer.hackathonwriteservice.hackathon.model.HackathonDto;
+import com.hackathonorganizer.hackathonwriteservice.hackathon.service.HackathonService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/v1/hackathons")
+@AllArgsConstructor
+public class HackathonController {
+
+    private final HackathonService hackathonService;
+
+    @PostMapping
+    @ResponseStatus(code = HttpStatus.CREATED)
+    public HackathonDto createHackathon(@RequestBody Hackathon hackathon) {
+
+        return hackathonService.createHackathon(hackathon);
+    }
+
+    @PatchMapping
+    public HackathonDto updateHackathonInfo(@RequestBody Hackathon hackathon) {
+
+        return hackathonService.updateHackathonData(hackathon);
+    }
+
+    @PatchMapping("/{id}/deactivate")
+    @ResponseStatus(code = HttpStatus.ACCEPTED)
+    public String deactivateHackathon(@PathVariable("id") Long hackathonId) {
+
+        return hackathonService.deactivateHackathon(hackathonId);
+    }
+
+    @PostMapping("/{id}/participants")
```

Comment:
Post jest uzywane gdy tworzysz nowy obiekt, proponuje tutaj patch lub put

Reply:
Patch ma użyte niżej do usuwania, nie wiem jaki był zamysł tego endpointa, czy to dla usera który ma się zapisywać na hackatony, czy dla organizatora do dopisywania osób. Raczej wywaliłbym te exceptiony z prób dodania/usunięcia jak pisałem w innym komenatrzu i połączyłbym te dwa endponty w jednego idempotętnego PUTa z userId + komenda ASSIGN/REMOVE, a jeśli endpoint dla organizatora to zdecydowanie szedłbym w operacje masowe bo na pewno na dodaniu jednego usera się nie skończy i bezsensownie będzie ten endpoint gwałcony dziesiątki razy